packages/frontend-app-api/src/wiring/createExtensionInstance.test.ts

```diff
@@ -250,6 +250,49 @@ describe('createExtensionInstance', () => {
     );
   });
 
+  it('should refuse to create an instance with undeclared inpurts', () => {
+    expect(() =>
+      createExtensionInstance({
+        attachments: new Map([
+          [
+            'declared',
+            [
+              createExtensionInstance({
+                attachments: new Map(),
+                config: { output: 'many1' },
+                extension: simpleExtension,
+              }),
+            ],
+          ],
+          [
+            'undeclared',
+            [
+              createExtensionInstance({
+                attachments: new Map(),
+                config: { output: 'many1' },
+                extension: simpleExtension,
+              }),
+            ],
+          ],
+        ]),
+        config: undefined,
+        extension: createExtension({
+          id: 'core.test',
+          attachTo: { id: 'ignored', input: 'ignored' },
+          inputs: {
+            declared: createExtensionInput({
+              test: testDataRef,
+            }),
+          },
+          output: {},
+          factory() {},
+        }),
+      }),
+    ).toThrow(
+      "Failed to instantiate extension 'core.test', received undeclared input(s) 'undeclared'",
```

Comment:
Hm. Am I reading this right that you refuse to instantiate A, because B tried to attach something in an unexpected place? Doesn't it reject B instead?

Reply:
It's a bit tricky to not have the extension being created be the point of throwing this error, but I updated it to include information about where the undeclared inputs are being attached from.